### ISSUE TITLE
rustfmt crashes

### ISSUE THREAD

**Issue by snsmac** [rust-lang/rustfmt#2848]
Hello,
[this file](https://pastebin.com/YAV2uC5s), crashes or hangs rust fmt.
With stable: rustfmt 0.6.1-stable (49279d7 2018-05-08) it crashes with following backtrace:
```
RUST_BACKTRACE=1 cargo +stable fmt
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', libcore/option.rs:335:21
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::print
             at libstd/sys_common/backtrace.rs:71
             at libstd/sys_common/backtrace.rs:59
   2: std::panicking::default_hook::{{closure}}
             at libstd/panicking.rs:211
   3: std::panicking::default_hook
             at libstd/panicking.rs:227
   4: std::panicking::rust_panic_with_hook
             at libstd/panicking.rs:463
   5: std::panicking::begin_panic_fmt
             at libstd/panicking.rs:350
   6: rust_begin_unwind
             at libstd/panicking.rs:328
   7: core::panicking::panic_fmt
             at libcore/panicking.rs:71
   8: core::panicking::panic
             at libcore/panicking.rs:51
   9: <rustfmt_nightly::overflow::Context<'a, T>>::new
  10: rustfmt_nightly::expr::rewrite_call
  11: rustfmt_nightly::chains::rewrite_chain_subexpr
  12: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T, I>>::from_iter
  13: rustfmt_nightly::chains::rewrite_chain
  14: rustfmt_nightly::expr::format_expr
  15: rustfmt_nightly::expr::<impl rustfmt_nightly::rewrite::Rewrite for syntax::ast::Stmt>::rewrite
  16: rustfmt_nightly::visitor::FmtVisitor::walk_stmts
  17: rustfmt_nightly::visitor::FmtVisitor::visit_block
  18: rustfmt_nightly::expr::rewrite_block_with_visitor
  19: rustfmt_nightly::expr::rewrite_block
  20: rustfmt_nightly::closures::rewrite_closure
  21: rustfmt_nightly::expr::format_expr
  22: <rustfmt_nightly::overflow::Context<'a, T>>::rewrite
  23: rustfmt_nightly::expr::rewrite_call
  24: rustfmt_nightly::chains::rewrite_chain_subexpr
  25: rustfmt_nightly::chains::rewrite_chain
  26: rustfmt_nightly::expr::format_expr
  27: rustfmt_nightly::expr::<impl rustfmt_nightly::rewrite::Rewrite for syntax::ast::Stmt>::rewrite
  28: rustfmt_nightly::visitor::FmtVisitor::walk_stmts
  29: rustfmt_nightly::visitor::FmtVisitor::visit_block
  30: rustfmt_nightly::expr::rewrite_block_with_visitor
  31: rustfmt_nightly::expr::rewrite_block
  32: rustfmt_nightly::closures::rewrite_closure
  33: rustfmt_nightly::expr::format_expr
  34: <rustfmt_nightly::overflow::Context<'a, T>>::rewrite
  35: rustfmt_nightly::expr::rewrite_call
  36: rustfmt_nightly::chains::rewrite_chain_subexpr
  37: rustfmt_nightly::chains::rewrite_chain
  38: rustfmt_nightly::expr::format_expr
  39: rustfmt_nightly::expr::<impl rustfmt_nightly::rewrite::Rewrite for syntax::ast::Stmt>::rewrite
  40: rustfmt_nightly::visitor::FmtVisitor::walk_stmts
  41: rustfmt_nightly::visitor::FmtVisitor::visit_block
  42: rustfmt_nightly::expr::rewrite_block_with_visitor
  43: rustfmt_nightly::expr::rewrite_block
  44: rustfmt_nightly::closures::rewrite_closure
  45: rustfmt_nightly::expr::format_expr
  46: <rustfmt_nightly::overflow::Context<'a, T>>::rewrite
  47: rustfmt_nightly::expr::rewrite_call
  48: rustfmt_nightly::chains::rewrite_chain_subexpr
  49: rustfmt_nightly::chains::rewrite_chain
  50: rustfmt_nightly::expr::format_expr
  51: rustfmt_nightly::expr::<impl rustfmt_nightly::rewrite::Rewrite for syntax::ast::Stmt>::rewrite
  52: rustfmt_nightly::visitor::FmtVisitor::walk_stmts
  53: rustfmt_nightly::visitor::FmtVisitor::visit_block
  54: rustfmt_nightly::expr::rewrite_block_with_visitor
  55: rustfmt_nightly::expr::rewrite_block
  56: rustfmt_nightly::closures::rewrite_closure
  57: rustfmt_nightly::expr::format_expr
  58: rustfmt_nightly::closures::rewrite_last_closure
  59: <rustfmt_nightly::overflow::Context<'a, T>>::rewrite
  60: rustfmt_nightly::expr::rewrite_call
  61: rustfmt_nightly::chains::rewrite_chain_subexpr
  62: rustfmt_nightly::chains::rewrite_chain
  63: rustfmt_nightly::expr::format_expr
  64: rustfmt_nightly::expr::<impl rustfmt_nightly::rewrite::Rewrite for syntax::ast::Stmt>::rewrite
  65: rustfmt_nightly::visitor::FmtVisitor::walk_stmts
  66: rustfmt_nightly::visitor::FmtVisitor::visit_block
  67: rustfmt_nightly::expr::rewrite_block_with_visitor
  68: rustfmt_nightly::expr::rewrite_block
  69: rustfmt_nightly::closures::rewrite_closure
  70: rustfmt_nightly::expr::format_expr
  71: <rustfmt_nightly::overflow::Context<'a, T>>::rewrite
  72: rustfmt_nightly::expr::rewrite_call
  73: rustfmt_nightly::chains::rewrite_chain_subexpr
  74: rustfmt_nightly::chains::rewrite_chain
  75: rustfmt_nightly::expr::format_expr
  76: rustfmt_nightly::closures::rewrite_closure_expr
  77: rustfmt_nightly::closures::rewrite_closure
  78: rustfmt_nightly::expr::format_expr
  79: rustfmt_nightly::closures::rewrite_last_closure
  80: <rustfmt_nightly::overflow::Context<'a, T>>::rewrite
  81: rustfmt_nightly::expr::rewrite_call
  82: rustfmt_nightly::chains::rewrite_chain_subexpr
  83: rustfmt_nightly::chains::rewrite_chain
  84: rustfmt_nightly::expr::format_expr
  85: rustfmt_nightly::expr::<impl rustfmt_nightly::rewrite::Rewrite for syntax::ast::Stmt>::rewrite
  86: rustfmt_nightly::visitor::FmtVisitor::walk_stmts
  87: rustfmt_nightly::visitor::FmtVisitor::visit_block
  88: rustfmt_nightly::expr::rewrite_block_with_visitor
  89: rustfmt_nightly::expr::rewrite_block
  90: rustfmt_nightly::closures::rewrite_closure
  91: rustfmt_nightly::expr::format_expr
  92: rustfmt_nightly::closures::rewrite_last_closure
  93: <rustfmt_nightly::overflow::Context<'a, T>>::rewrite
  94: rustfmt_nightly::expr::rewrite_call
  95: rustfmt_nightly::chains::rewrite_chain_subexpr
  96: rustfmt_nightly::chains::rewrite_chain
  97: rustfmt_nightly::expr::format_expr
  98: rustfmt_nightly::expr::<impl rustfmt_nightly::rewrite::Rewrite for syntax::ast::Stmt>::rewrite
  99: rustfmt_nightly::visitor::FmtVisitor::walk_stmts
```

Nightly (rustfmt 0.8.2-nightly (5e59925 2018-07-02)) seems to hang or is really slow

**Comment by topecongiro** [comment]
Still present in `rustfmt 1.0.1-nightly (be135599 2018-12-10)`. rustfmt takes more than 14 minutes to format the file.

**Comment by topecongiro** [comment]
Fixed as of `rustfmt 1.2.0-nightly (09940a7 2019-03-27)`.
